amps/ags/rm-automation/rm-automation-community-rest-api/src/main/java/org/alfresco/rest/v0/ExportAPI.java

```diff
@@ -0,0 +1,150 @@
+import java.util.List;
+
+import org.alfresco.rest.core.v0.BaseAPI;
+import org.alfresco.rest.rm.community.model.record.Record;
+import org.alfresco.rest.rm.community.model.recordcategory.RecordCategoryChild;
+import org.alfresco.rest.rm.community.model.recordfolder.RecordFolder;
+import org.alfresco.utility.report.log.Step;
+import org.apache.http.HttpResponse;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.stereotype.Component;
+
+/**
+ * Methods to make API requests using v0 API for Exporting a file
+ *
+ * @author Shubham Jain
+ * @since 3.5.0
+ */
+
+@Component
+public class ExportAPI extends BaseAPI
+{
+    /**
+     * The URI to export an item
+     */
+    private static final String EXPORT_API = "{0}rma/admin/export";
+
+    /**
+     * List to store the Node References of items to be exported
+     */
+    private static List<String> nodeRefs = new ArrayList<>();
+
+    /**
+     * Export a single Record using V0 Export API
+     *
+     * @param user               User performing the export
+     * @param password           User's Password
+     * @param expectedStatusCode Expected Response Code
+     * @param recordName         Name of the Record to be exported
+     * @return exportAPI function
+     */
+    public HttpResponse exportRecord(String user, String password, int expectedStatusCode, String recordName)
+    {
+        final String recNodeRef = getNodeRefSpacesStore() + contentService.getNodeRef(user, password, RM_SITE_ID,
+                recordName);
+
+        nodeRefs.add(recNodeRef);
+
+        return export(user, password, expectedStatusCode, nodeRefs);
+    }
+
+    /**
+     * Export an array of records using V0 Export API
+     *
+     * @param user               User performing the export
+     * @param password           User's Password
+     * @param expectedStatusCode Expected Response Code
+     * @param recordList         List of the records to be exported
+     * @return exportAPI function
+     */
+    public HttpResponse exportRecords(String user, String password, int expectedStatusCode, List<Record> recordList)
+    {
+
+        for (int i = 0; i < recordList.size(); i++)
+        {
+            String recNodeRef = getNodeRefSpacesStore() + contentService.getNodeRef(user, password, RM_SITE_ID,
+                    recordList.get(i).getName());
+
+            nodeRefs.add(recNodeRef);
+        }
+
+        return export(user, password, expectedStatusCode, nodeRefs);
+    }
+
+    /**
+     * Export a Record Folder containing records using V0 Export API
+     *
+     * @param user               User performing the export
+     * @param password           User's Password
+     * @param expectedStatusCode Expected Response Code
+     * @param folderName         Name of the Record Folder to be exported
+     * @return exportAPI function
+     */
+    public HttpResponse exportRecordFolder(String user, String password, int expectedStatusCode, String folderName)
```

Comment:
This method is duplicate of exportRecord, only the item is different (rec folder, instead of record) .Please note that you should be able to export also categories. In this case I propose to rename the method exportRecord to be exportNode or exportItem and to be used for whatever kind of item that you need.

Reply:
Noted